builder.go

```diff
@@ -44,10 +56,48 @@ func WithCustomHTTPRoundTripper(r http.RoundTripper) Option {
 	}
 }
 
+// WithMiddlewareAfter ...
+func WithMiddlewareAfter(after ...AfterExecute) Option {
+	return func(o *options) {
+		o.middleware.After = append(o.middleware.After, after...)
+	}
+}
+
+// WithMiddlewareAfterT ...
+func WithMiddlewareAfterT(after ...AfterExecuteT) Option {
+	return func(o *options) {
+		o.middleware.AfterT = append(o.middleware.AfterT, after...)
+	}
+}
+
+// WithMiddlewareBefore ...
+func WithMiddlewareBefore(before ...BeforeExecute) Option {
+	return func(o *options) {
+		o.middleware.Before = append(o.middleware.Before, before...)
+	}
+}
+
+// WithMiddlewareBeforeT ...
+func WithMiddlewareBeforeT(beforeT ...BeforeExecuteT) Option {
+	return func(o *options) {
+		o.middleware.BeforeT = append(o.middleware.BeforeT, beforeT...)
+	}
+}
+
+// WithHardValidation ...
+func WithHardValidation() Option {
+	return func(o *options) {
+		o.hardValidation = true
+	}
+}
+
 // NewHTTPTestMaker is function for set options for all cute.
 func NewHTTPTestMaker(opts ...Option) *HTTPTestMaker {
 	var (
-		o            = new(options)
+		o = &options{
```

Comment:
perhaps it would be better to create an object without a reference, and then pass a reference to it